gobblin-data-management/src/main/java/gobblin/data/management/policy/NewestKSelectionPolicy.java

```diff
@@ -28,30 +29,89 @@
 /**
  * Select the newest k versions of the dataset.
  */
-public class NewestKSelectionPolicy<T extends DatasetVersion> implements VersionSelectionPolicy<T> {
+public class NewestKSelectionPolicy implements VersionSelectionPolicy<DatasetVersion> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(NewestKSelectionPolicy.class);
 
+  /**
+   * The number of newest versions to select. Only one of
+   * {@link #NEWEST_K_VERSIONS_SELECTED_KEY} and {@link #NEWEST_K_VERSIONS_NOTSELECTED_KEY} can
+   * be specified. The default is {@link #NEWEST_K_VERSIONS_SELECTED_KEY} with a value of
+   * {@link #VERSIONS_SELECTED_DEFAULT_INT}. Valid values are in the range
+   * [{@link #MIN_VERSIONS_ALLOWED}, {@link #MAX_VERSIONS_ALLOWED}]. Negative values are interpreted
+   * as {@link #NEWEST_K_VERSIONS_NOTSELECTED_KEY} with the absolute value.
+   */
   public static final String NEWEST_K_VERSIONS_SELECTED_KEY = "selection.newestK.versionsSelected";
 
-  public static final String VERSIONS_SELECTED_DEFAULT = Integer.toString(2);
+  /**
+   * The number of newest versions to exclude from the result. Only one of
+   * {@link #NEWEST_K_VERSIONS_SELECTED_KEY} and {@link #NEWEST_K_VERSIONS_NOTSELECTED_KEY} can
+   * be specified. The default is {@link #NEWEST_K_VERSIONS_SELECTED_KEY} with a value of
+   * {@link #VERSIONS_SELECTED_DEFAULT}. Valid values are in the range
+   * [{@link #MIN_VERSIONS_ALLOWED}, {@link #MAX_VERSIONS_ALLOWED}]. Negative values are interpreted
```

Comment:
I am not completely sure about this. Was there a reason to support Negative values for these keys? This would mean there are 4 ways to configure this class
versionsSelected : positive k
versionsSelected : negative k
versionsExcluded : positive k
versionsExcluded : negative k

2,3 mean the same and 1,4 mean the same. Since the desired behavior can be achieved by using positive k's only, I think we should restrict users to specify positive value. We will have fewer code branches.

<pre>
 public Collection<DatasetVersion> listSelectedVersions(List<DatasetVersion> allVersions) {
  if (versionsSelected > 0) {
       return boundarySafeSublist(allVersions, 0, versionsSelected);
  } else {
       return boundarySafeSublist(allVersions, versionsExcluded, allVersions.size());
  }
}
</pre>
What do you think?